lib/access-context.js

```diff
@@ -12,17 +12,29 @@ var debug = require('debug')('loopback:security:access-context');
  * Access context represents the context for a request to access protected
  * resources
  *
+ * NOTE While the method expects an array of principals in the AccessContext instance/object,
+ * it also accepts a single principal defined with the following properties:
+ * ```js
+ * {
+ *   // AccessContext instance/object
+ *   // ..
+ *   principalType: 'somePrincipalType', // APP, ROLE, USER, or custom user model name
+ *   principalId: 'somePrincipalId',
+ * }
+ * ```
+ *
```

Comment:
adding a comment to document the ability to pass a principal definition at root of the AccessContext instance
I did not referenced the [prinicipalName](https://github.com/strongloop/loopback/blob/master/lib/access-context.js#L67) property as i can't seem to find any usage of it in the entire codebase